File: C++/Greedy/jump-game-55.cpp
// Greedy Choice - [Leetcode 55](https://leetcode.com/problems/jump-game/description/)

#include <vector>
using namespace std;

/*
    Greedy Choice -> Take the fartest jump and see if it reaches the end
*/
bool canJump(vector<int>& nums) {
    int loc = 0;
    int n = nums.size();
    for ( int i =0; i < n; i++ ) {
        if ( loc < i || loc >= n - 1 ) {
            break;
        }
        loc = max(loc, i + nums[i]);
    }
    return loc >= n - 1;
}

File: Python/Greedy/jump-game-55.py
# Max Jump as Greedy Choice - [Leetcode 55](https://leetcode.com/problems/jump-game/)
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxJump = 0
        for idx, val in enumerate(nums):
            if maxJump < idx:
                return False
            maxJump = max(maxJump, idx + val)
        return True
